lib/logstash/plugin_mixins/snmp/common.rb

```diff
@@ -118,7 +118,7 @@ def build_snmp_client!(client_builder, validate_usm_user: false)
           validate_usm_user! if validate_usm_user
 
           unless @security_name.nil?
-            client_builder.addUsmUser(@security_name, @auth_protocol, @auth_pass&.value, @priv_protocol, @priv_pass&.value)
+            client_builder.addUsmUser(@security_name, @auth_protocol, @auth_pass&.value, @priv_protocol, @priv_pass&.value, @security_level || 'noAuthNoPriv')
```

Comment:
🤔 if the user provides a `security_name` and does _not_ provide a `security_level`, we now enforce `noAuthNoPriv`, and therefore silently reject trapped messages that _do_ have authentication and/or privacy matching the provided `auth_*` and/or `priv_*` configuration. This feels like a breaking change to me.

Reply:
resolved